### PR TITLE
POM for check access (Access and Submission) page

### LINKED ISSUE #3591: POM for check access
linked to #3485

### PR DESCRIPTION
**Pull request by eazapata23**
This PR is linked to #3485
This PR will close #3591 

In this PR:

- I added a Page objects page for `check-access.js`
- Removed `check-access.js` from support
- Modified `initialize-audit.js` to use new import and to instantiate an object from `accessAndSubmissionPage` class that calls the method `testValidAccess()`

# How to test:

- Run cypress as you normally would locally and run `full-submission.cy.js` test

## PR checklist: submitters

- [x]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [x]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [x]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [x]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [x]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/cypress/e2e/createNewAudit.cy.js [hunk 1]
```diff
@@ -37,6 +37,6 @@ describe('Create new audit', () => {
   });
 
   it('Fills out the General Information Form', () => {
-    testValidGeneralInfo();
+    GeneralInfoForm();
   });
-});
+});
```
Comment (phildominguez-gsa) [inline backend/cypress/e2e/createNewAudit.cy.js:42]: I'd put this newline back so GH stops complaining. Some of your other files are missing EoF newlines as well. [Here's](https://stackoverflow.com/a/729795) why they're good.

File: backend/cypress/e2e/search-tests.cy.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    //search results
+    search.testSearchTable();
+
+    //summary report
+    search.testSummaryReport();
+
+  });
+
+
+});
```
Comment (phildominguez-gsa) [inline backend/cypress/e2e/search-tests.cy.js:95]:
```suggestion
    search.testSummaryReport();
  });

});
```

File: backend/cypress/e2e/search-tests.cy.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+//advanced search test
```
Comment (phildominguez-gsa) [inline backend/cypress/e2e/search-tests.cy.js:97]: I think either remove this commented out stuff, or put a comment explaining why it's commented out.

File: backend/cypress/pageObjects/general-info.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import auditorCityField from '../fixtures/genInfoData/auditorCityField.json';
+import auditorStateField from '../fixtures/genInfoData/auditorStateField.json';
+import auditorZipField from '../fixtures/genInfoData/auditorZipField.json';
+import auditorContactNameField from '../fixtures/genInfoData/auditorContactNameField.json';
+import auditorContactTitleField from '../fixtures/genInfoData/auditorContactTitleField.json';
+import auditorPhoneField from '../fixtures/genInfoData/auditorPhoneField.json';
+import auditorEmailField from '../fixtures/genInfoData/auditorEmailField.json';
+
+
+class GeneralInfoForm {
```
Comment (phildominguez-gsa) [inline backend/cypress/pageObjects/general-info.js:23]:
```suggestion
import auditorEmailField from '../fixtures/genInfoData/auditorEmailField.json';

class GeneralInfoForm {
```

File: backend/cypress/pageObjects/general-info.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        this.auditorPhoneField = '#auditor_phone';
+        this.auditorEmailField = '#auditor_email';
+        this.continueButton = '#continue';
+        this.auditeeUEI = '#auditee_uei';
+        this.multipleUEIsYesLabel = 'label[for=multiple-ueis-yes]';
+        this.secondaryAuditorsField = 'label[for=secondary_auditors-yes]';
+    }
+    
+
+    selectAuditTypeAndPeriod() {
```
Comment (phildominguez-gsa) [inline backend/cypress/pageObjects/general-info.js:57]:
```suggestion
    }

    selectAuditTypeAndPeriod() {
```

File: backend/cypress/pageObjects/general-info.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    clickContinueButton() {
+        cy.get(this.continueButton).click();
+        cy.url().should('match', /\/audit\/submission-progress\/[0-9]{4}-[0-9]{2}-GSAFAC-[0-9]{10}/);
+    }
+    
+    
+}
+
+
+export default GeneralInfoForm;
```
Comment (phildominguez-gsa) [inline backend/cypress/pageObjects/general-info.js:103]:
```suggestion
    }
}

export default GeneralInfoForm;
```

File: backend/cypress/pageObjects/searchPage.js [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        const reportId = url.split('/').pop();
+        const downloadUrl = Cypress.config().baseUrl + `/dissemination/summary/${reportId}`
+        cy.downloadFile(downloadUrl,
+            'cypress/downloads', '2022-12-GSAFAC-0000000001.pdf');
+    }
+
+
+}
+
+export default searchPage;
```
Comment (phildominguez-gsa) [inline backend/cypress/pageObjects/searchPage.js:163]:
```suggestion
    }
}

export default searchPage;

```

**Review by phildominguez-gsa — CHANGES_REQUESTED** [review]
Works nicely! Just smallish stuff. One other thing to keep in mind keeping a consistent casing style for folders and files. From looking around we seem to use `snake_case` for folders and `kebab-case` for js and json files.

**Comment by eazapata23** [comment]
this PR is no longer needed and will be closed since the work has been done in this PR #3574

**Pull request closed without merging**